test/router-luatest/router_test.lua

```diff
@@ -285,6 +285,190 @@ g.test_map_callrw_raw = function(g)
     end)
 end
 
+g.test_group_buckets = function(g)
+    local bids = vtest.storage_get_n_buckets(g.replica_1_a, 2)
+
+    local res = g.router:exec(function(bid1, bid2)
+        local val, err = ivshard.router.group({bid2, bid1, bid1})
+        return {
+            val = val,
+            err = err,
+        }
+    end, {bids[1], bids[2]})
+    assert(res.err == nil)
+    local rs1_uuid = g.replica_1_a:replicaset_uuid()
+    local expected = {
+        [rs1_uuid] = {bids[1], bids[2]},
+    }
+    table.sort(expected[rs1_uuid])
+    t.assert_equals(res.val, expected)
+end
+
+g.test_map_part_callrw = function(g)
```

Comment:
Please, split it into a couple of separate test cases. One would be `test_map_part_callrw_basic()` (with lines 308-364 in your latest version of the patch and "Error at map stage" case), another - `test_map_part_callrw_cant_ref()` (with the test for SENT buckets timing out the refs).

Reply:
I have separated the tests and moved them to a separate file (router_test.lua looks too big with map_part tests)